### ISSUE TITLE
Bounding boxes not tight/inaccurate when using yolov8s-obb

### ISSUE THREAD

**Issue by farhanrw** [ultralytics/ultralytics#9209]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, firstly thanks to the whole team for the amazing work in developing yolov8 (especially with OBB). I have been working with the regular detection model, and over the last few days, I have been playing around with the OBB models, yolov8s-obb to be specific. I noticed an issue - so when I have a few degrees of rotation in my augmentation file (i.e. degrees: 30.0), the train_batch images that get written out in the trains folder have images with invalid and missing overlays. I am attaching two images as an example. Without rotation augmentation, I can constantly see the overlays are not tightly fitting around the images in all cases. By cases I mean from what I see in train_batch0,1 and 2 inside the train folder. 

To illustrate this problem clearly, I created a dummy dataset of 4-5 images with a colorful rectangle sitting at the center and labeled the images. I labeled these myself and exported using Roboflow and for Yolov8-obb. As you can see in the images, the first image is without using rotation augmentation and you can still see the ROIs drawn by yolov8 not fitting the box tightly, although my labels were accurate.

![Capture1](https://github.com/ultralytics/ultralytics/assets/43019847/d86a340c-abc1-459e-b6c2-edb3d4021a56)

In the second image, you can see how one of them does have the bounding box but the orientation is not accurate (the center orange one and the top-rightmost one).

![Capture](https://github.com/ultralytics/ultralytics/assets/43019847/0fe37271-dd2d-45d0-b1fb-8700c42f7612)

Unfortunately, I could not repeat missing bounding box cases to show here, but I will add it here if I see it randomly pop up. Thanks and any help would be great!

**Comment by glenn-jocher** [comment]
@farhanrw hello! 👋 Thanks for reaching out and sharing your detailed observations with us.

For the issue of not-so-tight bounding boxes with `yolov8s-obb`, especially when incorporating rotation augmentation, the behavior you've mentioned can sometimes result from model predictions that don't perfectly align with the actual object orientations or sizes in augmented conditions. This could be influenced by several factors, including the model's training dataset diversity, augmentation parameters, and the inherent challenges in accurately predicting oriented bounding boxes (OBBs).

To potentially improve the tightness of the OBBs, you might consider:

1. Increasing dataset diversity with more varied orientations and sizes of your objects of interest.
2. Adjusting the augmentation parameters gradually to see if there's a sweet spot that improves the bounding box fit. For instance, you could experiment with different degrees of rotation:
   ```yaml
   degrees: 15.0 # trying a smaller rotation degree
   ```
3. Reviewing the training loss and accuracy metrics to ensure the model is not underfitting or overfitting.

For the cases of missing bounding boxes, ensuring that the dataset contains enough examples of similar object orientations and configurations that might appear due to augmentation could help the model learn to detect these scenarios more reliably.

If these suggestions don't resolve the issue, it might be valuable to further investigate the model's predictions on a broader set of samples or consider re-training the model with an adjusted dataset or augmentation strategy. 

Your initiative in testing and providing feedback is greatly appreciated, as it helps us improve. If you have further insights or questions, feel free to share! 😊

**Comment by farhanrw** [comment]
Hi, @glenn-jocher these images are not test sets or predictions. These are the training batch images that get written out in the train directory at the start of the training. That is where my concern lies because my ground truth label for the training data certainly does not look like this. Again, these are not prediction overlays rather the train batch 0,1,2… images.

**Comment by glenn-jocher** [comment]
@farhanrw hi there! Thanks for the clarification, and for bringing this to our attention. It sounds like there's an inconsistency in how the training batches are visualized, especially with your ground truth data not matching the displayed overlays in the training directory. 

This might relate to how the augmentation, particularly the rotation, is being applied or visualized during the training setup stage. It's worth checking if the augmentation parameters in your YAML file are being interpreted as intended. For example, ensuring the `degrees` parameter for rotation augmentation is correctly set as follows:

```yaml
# In your dataset YAML
augment:
  degrees: 30.0  # Or any other value you're using
```

It could also be helpful to verify the visual representation script to ensure that it correctly projects the ground truth labels, post-augmentation, onto these batch images.

If everything seems correct on your end and the issue persists, we might need to dive deeper into this. Would be great if you could share more details or examples so we can further investigate together. 🕵️

Thanks for your patience and assistance in troubleshooting this!

**Comment by mudassir659** [comment]
@glenn-jocher please help to change my bounding box width in yolov8 how can i put it directly on my cli command while doing detection i am not getting results by using this code yolo task=detect mode=predict model=** conf=0.7 source=*** project=** save=True line_width=none

**Comment by farhanrw** [comment]
Hi, @glenn-jocher yes I verified all those things you mentioned, but did another round of checks. I visualized the exported labels in Roboflow and then also used my own simple script to take in the labels and draw custom overlays using OpenCV and all worked out well. More importantly, if I use the regular yolov8 models, even if I use rotations, the training batch images are drawn just fine. This issue only shows up when using the OBB models. Again, if you look at the images I attached, there are 2/3 similar but different problems. 1. No rotation - Loose/inaccurate boxes  2. With rotation augmentation (10-30 degrees) = Inaccurate boxes + Missing boxes. I attached images in my original post above. I can provide more if needed. Thanks!

**Comment by glenn-jocher** [comment]
Hi @farhanrw, thanks for the detailed follow-up and for checking everything so thoroughly! 🙌 It's helpful to know that this issue is specific to the OBB models and doesn't occur with the regular YOLOv8 models. This indeed hints at something unique to the OBB implementation or its interaction with rotations.

Given the inconsistencies you're observing with the OBB models, especially after confirmation that your labels are accurately drawn, this could be an area for us to investigate further internally. The distinctions you've noted between scenarios with and without rotation, and the specific issues tied to the OBB models, provide a good starting point for us.

If it's not too much trouble, additional examples illustrating the problems would indeed be beneficial for a deeper analysis. Meanwhile, we'll begin to look into this peculiarity with OBB model handling, especially under rotation augmentation settings.

Thanks so much for bringing this to our attention and for your willingness to provide more insights if needed. Your input is invaluable in helping us improve. Stay tuned! 🤓

**Comment by Sagi-Beewise** [comment]
This problem has also been encountered by me.
Are there any updates to the yolov8-obb repository regarding this issue?

**Comment by glenn-jocher** [comment]
Hi @Sagi-Beewise, thanks for reaching out and for your patience! 😊

We appreciate you bringing this to our attention. To help us investigate and address this issue effectively, could you please provide a minimum reproducible code example? This will allow us to replicate the problem on our end. You can find guidelines on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example). This step is crucial for us to understand the specifics of the issue and work towards a solution.

Additionally, please ensure that you are using the latest versions of `torch` and `ultralytics`. Sometimes, updates include important bug fixes and improvements that might resolve the issue you're encountering. You can upgrade your packages using the following commands:

```bash
pip install --upgrade torch
pip install --upgrade ultralytics
```

If the problem persists after updating, please share the reproducible code example along with any relevant details about your environment and dataset. This will greatly assist us in diagnosing and addressing the issue.

Thank you for your cooperation and understanding. We're here to help and look forward to resolving this with you! 🚀

**Comment by farhanrw** [comment]
> This problem has also been encountered by me. Are there any updates to the yolov8-obb repository regarding this issue?

Hi, @Sagi-Beewise I have raised another issue related to this (#10181 ), maybe you are facing the same problem as well.
